core/src/main/scala/kafka/server/DelayedOperation.scala

```diff
@@ -289,35 +296,35 @@ class DelayedOperationPurgatory[T <: DelayedOperation](purgatoryName: String,
    * A linked list of watched delayed operations based on some key
    */
   private class Watchers(val key: Any) {
+    private[this] val operations = new ConcurrentLinkedQueue[T]()
 
-    private[this] val operations = new LinkedList[T]()
+    // note that size() is O(n) for ConcurrentLinkedQueue. Use isEmpty() if possible.
+    def watched: Int = operations.size
```

Comment:
Nit: what do you think of my suggestion to call this `countWatched` or something like that to make it clearer that it's not just a field that is returned?